atom/renderer/api/atom_api_spell_check_client.cc

```diff
@@ -163,12 +172,20 @@ bool SpellCheckClient::IsValidContraction(const base::string16& contraction) {
   int word_length;
 
   while (contraction_iterator_.GetNextWord(&word, &word_start, &word_length)) {
-    if (!SpellCheckWord(word))
+    if (!SpellCheckWord(scope, word))
       return false;
   }
   return true;
 }
 
+SpellCheckClient::SpellCheckScope::SpellCheckScope(
+    const SpellCheckClient& client)
+    : handle_scope_(client.isolate_),
+      context_scope_(
+          v8::Local<v8::Context>::New(client.isolate_, client.context_)),
+      provider_(client.provider_.NewHandle()),
+      spell_check_(client.spell_check_.NewHandle()) {}
```

Comment:
I don't understand the native_mate/v8 bridge enough to know if this is safe & defer that to wiser heads :)

_If_ it is safe, +1 on doing the `v8::Local` conversion once instead of each time `SpellCheckWord()` is called

Reply:
@zcbenz or @deepak1556 could you do a review on this?

Reply:
LGTM 👍